### PR TITLE
Fixed #1665: Default to ECMAScript 2015

### LINKED ISSUE #1665: Scala.js: Default to ECMAScript 2015
The official default for Scala.js is ECMAScript 2015, but Mill currently defaults to ES 5.1, here:

```scala
def useECMAScript2015: T[Boolean] = false
```

https://github.com/com-lihaoyi/mill/blob/main/scalajslib/src/ScalaJSModule.scala#L200

Mill should default to ES 2015 instead. The workaround is to add this line to your `build.sc` file:

```scala
override def useECMAScript2015 = T(true)
```

### PR DESCRIPTION
**Pull request by davesmith00000**
Fix for https://github.com/com-lihaoyi/mill/issues/1665

### CODE REVIEW

**Review by lolgab — APPROVED** [review]
LGTM!

File: scalajslib/src/ScalaJSModule.scala [hunk 1]
```diff
@@ -197,7 +197,7 @@ trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
   def moduleKind: T[ModuleKind] = T { ModuleKind.NoModule }
 
-  def useECMAScript2015: T[Boolean] = false
+  def useECMAScript2015: T[Boolean] = true
```
Comment (lefou) [inline scalajslib/src/ScalaJSModule.scala:200]: As suggested in issue #1665, enable ECMAScript2015 only for ScalaJS 1.x.

```suggestion
  def useECMAScript2015: T[Boolean] = T { 
    !scalaJSVersion().startsWith("0.")
  }
```

Comment (davesmith00000) [reply scalajslib/src/ScalaJSModule.scala]: Looks good - thanks!

**Comment by lefou** [comment]
Thank you!

**Pull request merged**